client/state/notices/actions.js

```diff
@@ -38,6 +41,13 @@ function createNotice( status, text, options = {} ) {
 	}
 }
 
+export function setRoute( path ) {
```

Comment:
And I agree, I would like this to be somewhere else, but first we need to tackle routing in redux, which I would like to see :)
But for now, it makes sense here IMO. If we have more use cases for this kind of action, I will switch to the proper action from a proper tree.